istioctl/cmd/webhook.go

```diff
@@ -0,0 +1,510 @@
+				return err
+			}
+			webhookConfig, err := buildValidatingWebhookConfig(
+				caCert,
+				opts.webhookConfigPath,
+				opts.validatingWebhookConfigName,
+			)
+			if err != nil {
+				fmt.Printf("err when build validatingwebhookconfiguration: %v\n", err)
+				return err
+			}
+			whRet, err := createValidatingWebhookConfig(client, webhookConfig)
+			if err != nil {
+				fmt.Printf("error when creating validatingwebhookconfiguration: %v\n", err)
+				return err
+			}
+			if whRet == nil {
+				fmt.Println("validatingwebhookconfiguration created is nil")
+				return fmt.Errorf("validatingwebhookconfiguration created is nil")
+			}
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVar(&opts.enableValidationWebhook, "validation", true, "Specifies whether enabling"+
+		"the validating webhook.")
+	// Specifies the local file path to webhook CA certificate.
+	// If empty, will read CA certificate from k8s service account.
+	flags.StringVar(&opts.caCertPath, "ca-cert-path", "",
+		"Specifies the local file path to webhook CA certificate. If your cluster has custom k8s CA siging "+
+			"certificate --cluster-signing-cert-file, you need to specify this parameter.")
+	flags.StringVar(&opts.webhookConfigPath, "config-path", "",
+		"Specifies the file path of the webhook configuration.")
+	flags.StringVar(&opts.validatingWebhookConfigName, "config-name", "istio-galley",
+		"The name of the ValidatingWebhookConfiguration to manage.")
+	flags.StringVar(&opts.validatingWebhookServiceName, "service", "istio-galley",
+		"The service name of the validating webhook to manage.")
+	flags.StringVar(&opts.webhookSecretName, "webhook-secret-name", "",
+		"The name of a webhook secret in the cluster. istioctl will check whether the webhook certificate "+
+			"in the secret is issued by the CA certificate in the k8s service account or the given CA certificate, if any.")
+	flags.StringVar(&opts.webhookSecretNameSpace, "webhook-secret-namespace", "",
+		"The namespace of a webhook secret in the cluster, which is used together with --webhook-secret-name.")
+	flags.IntVar(&opts.maxTimeForCheckingWebhookServer, "timeout", 60,
+		"	Max time (in seconds) for checking the validating webhook server. If the validating webhook server is not ready"+
+			"in the given time, exit. Otherwise, apply the webhook configuration.")
+	flags.IntVar(&opts.maxTimeForReadingWebhookCert, "read-cert-timeout", 60,
+		"	Max time (in seconds) for waiting the webhook certificate to be readable.")
+
+	return cmd
+}
+
+func newDisableCmd(opts *disableCliOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "disable",
+		Short: "Disable webhook configurations",
+		Example: `
+# Disable the webhook configuration of Galley
+istioctl experimental post-install webhook disable --validation --config-name istio-galley
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !opts.disableValidationWebhook {
+				fmt.Println("not disabling validation webhook")
```

Comment:
Throughout, we should return the error instead of `fmt.Println` and returned nil. The caller will print the error message and return a non-zero exit code to the caller as expected.

Reply:
Done.